Sources/Host/Darwin/LibProc.cpp

```diff
@@ -55,7 +58,23 @@ void LibProc::EnumerateProcesses(
 }
 
 std::string LibProc::GetThreadName(ProcessId pid, ThreadId tid) {
-  return "<unknown>";
+  std::string dft("<unknown-thread>");
+  thread_identifier_info_data_t threadId;
+  struct proc_threadinfo ti;
+  ErrorCode err;
+  Mach mach;
+  int res;
+
+  err = mach.getThreadIdentifierInfo(pid, tid, &threadId);
+  if (err != kSuccess)
+    return dft;
+
+  res = proc_pidinfo(pid, PROC_PIDTHREADINFO, threadId.thread_handle, &ti,
+                     sizeof(ti));
+  if (res <= 0)
+    return dft;
+
+  return std::string(ti.pth_name);
```

Comment:
The constructor for `std::string` is implicit, I think you can just do `return ti.pth_name;`